### ISSUE TITLE
Redefined Gateway API Types

### ISSUE THREAD

**Issue by danehans** [kubernetes-sigs/gateway-api-inference-extension#115]
Currently, gateway-api-inference-extension defines API types that exist in Gateway API. For example:

```go
// PoolObjectReference identifies an API object within the namespace of the
// referrer.
type PoolObjectReference struct {
	Group string `json:"group,omitempty"`
	Kind  string `json:"kind,omitempty"`
	Name  string `json:"name,omitempty"`
}
```

Gateway API defines custom alias types for `Group`, `Kind`, and `Name`. For consistency, should these types be imported and used instead of redefining them as opaque strings? For example:

```go
import gwv1 "sigs.k8s.io/gateway-api/apis/v1"

// PoolObjectReference identifies an API object within the namespace of the
// referrer.
type PoolObjectReference struct {
	Group gwv1.Group `json:"group,omitempty"`
	Kind  gwv1.Kind `json:"kind,omitempty"`
	Name  gwv1.ObjectName `json:"name,omitempty"`
}
```

**Comment by danehans** [comment]
cc: @robscott

**Comment by robscott** [comment]
Good catch, thanks @danehans! The goal here was to add some defaults here so 99% of users would only need to specify a name, similar to how BackendRef works in Gateway API. Those have been added now, so hopefully it makes a bit more sense:

https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/60bab3a7c0da2eb1d9044cc6d9f779c3f794cf6c/api/v1alpha1/inferencemodel_types.go#L71-L89

**Comment by danehans** [comment]
@robscott thanks for the feedback. Even with defaults and validation added to the API fields, I'm trying to understand why concrete types are used here. In contrast, alias types are used in Gateway API object reference types. For example `Group string` here vs `Group *Group` in Gateway API [BackendObjectReference](https://github.com/kubernetes-sigs/gateway-api/blob/v1.2.1/apis/v1/object_reference_types.go#L102).

**Comment by robscott** [comment]
@danehans Got it, somehow I completely missed that part of the difference. Agree that it would make sense to reuse gwv1.Group, Kind, and ObjectName types, thanks for catching this!

**Comment by ahg-g** [comment]
My worry is about creating an unnecessary dependency on the gateway api pkg types. The group/kind of CRDs are not a gateway concept, they are a k8s one and in there they are just defined as strings: https://github.com/kubernetes/kubernetes/blob/6746df77f2376c6bc1fd0de767d2a94e6bd6cec1/pkg/apis/core/types.go#L5750.

**Comment by danehans** [comment]
I think it's fair to not not create Gateway API dependencies. Maybe this should be documented somewhere to help guide API development for the project. Thoughts?

**Comment by robscott** [comment]
Yeah, that's a good idea, I'll try to add that into the developer guide in https://github.com/kubernetes-sigs/gateway-api-inference-extension/pull/129.

**Comment by danehans** [comment]
@robscott can you confirm ^ was implemented in #129? If so, can you close this issue?

**Comment by ahg-g** [comment]
Closing this as the api merged already
